google-cloud-firestore/src/main/java/com/google/cloud/firestore/BulkWriter.java

```diff
@@ -85,8 +117,11 @@ boolean allowDuplicateDocs() {
   /** The maximum number of writes that can be in a single batch. */
   private int maxBatchSize = MAX_BATCH_SIZE;
 
-  /** A queue of batches to be written. */
-  private final List<BulkCommitBatch> batchQueue = new ArrayList<>();
+  /**
+   * A queue of batches to be written. Use a synchronized list to avoid multi-thread concurrent
+   * modification errors.
+   */
```

Comment:
```suggestion
  /**
   * A queue of batches to be written. Use a synchronized list to avoid multi-thread concurrent
   * modification errors (as this list is modified from both the user thread and the network thread).
   */
```